### PR TITLE
Feature :watch Later

### PR DESCRIPTION
**Pull request by haarsh24**
1. User can add video to watch later by clicking on watch later icon
2. User can go on watch later screen and see all the videos they have selected
3. user can remove videso from watch later by clicking on watch later icon from watch later page.

### CODE REVIEW

File: src/components/videoCard/videoCard.jsx [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
   const { videoList } = useVideo();
-  console.log(videoList);
+  const { watchLaterDispatch } = useWatchLater();
+  
+  const watchLaterHandler = () => {
+    watchLaterClicked ? setWatchLaterClicked(false) : setWatchLaterClicked(true);
+    watchLaterDispatch({
+                 type: "ADD_VIDEO_TO_WATCH_LATER",
+                 payload: videoData,
+               })
```
Comment (prachi-sahani) [inline src/components/videoCard/videoCard.jsx]: With this logic, if I click multiple times on the the watchlater icon, that video will get added multiple times to the watch later videos list. You can give a condition, if the video is already in the watch later list(or if icon is already clicked) then the ```watchLaterDispatch``` should be called with type ```"REMOVE_VIDEO_FROM_WATCH_LATER"```.
Attaching a screenshot for your reference
image

**Review by prachi-sahani — COMMENTED** [review]
Overall the video is clean and easy to understand. You may consider modifying the logic as mention in the comment.

**Suggestions:**
- you may add proper messages when there are no videos added.
- minor css issue as shown in screenshot
image

- once i add anything to watch later, navigate to some other tab and come back to video listing page, the watch later status is not retained(if i have already added to watch list, it must show the icon with primary color)
- watch later icon is not visible when the image has white backgound(suggested solution - you can keep the watch later icon in the title section)

image

File: src/components/videoCard/videoCard.jsx (hunk 1 above)
Comment (haarsh24) [reply src/components/videoCard/videoCard.jsx]: Thank you for your valuable feedback :
Will definitely make the suggested changes.

**Review by yashpatel1797 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

**Pull request merged**